tests/sentry/notifications/api/endpoints/test_notification_actions_details.py

```diff
@@ -470,3 +470,84 @@ def test_delete_team_admin(self):
         self.login_as(user)
 
         self.test_delete_simple()
+
+    def test_get_respects_multiple_project_access(self):
+        # Disable open membership
+        self.organization.flags.allow_joinleave = False
+        self.organization.save()
+        # Create a user, team, project
+        user = self.create_user()
+        self.create_member(user=user, organization=self.organization)
+        team = self.create_team(name="mobile", organization=self.organization, members=[user])
+        access_project = self.create_project(
+            organization=self.organization, teams=[team], name="ionia"
+        )
+        # Attempt to access a different project's action, should fail since open membership is off
+        self.login_as(user)
+        self.get_error_response(
+            self.organization.slug,
+            self.notif_action.id,
+            status_code=status.HTTP_403_FORBIDDEN,
+            method="GET",
+        )
+        # Add the project to the action to allow it to succeed now
+        self.notif_action.projects.add(access_project)
+        self.notif_action.save()
+        self.get_success_response(
+            self.organization.slug,
+            self.notif_action.id,
+            status_code=status.HTTP_200_OK,
+            method="GET",
+        )
+
+    def test_delete_respects_multiple_project_access(self):
+        # Disable open membership
+        self.organization.flags.allow_joinleave = False
+        self.organization.save()
+        # Create a user, team, project
+        user = self.create_user()
+        member = self.create_member(user=user, organization=self.organization)
+        team = self.create_team(name="mobile", organization=self.organization)
+        # Unrelated team, shouldn't affect anything
+        self.create_team(name="desktop", organization=self.organization, members=[user])
```

Comment:
This addition should supplement `test_delete_when_team_level_roles` from https://github.com/getsentry/sentry/pull/94946/files